### PR TITLE
New documentation

### LINKED ISSUE #853: Find alternative for Adobe Caslon Pro
Building the manual to cut releases requires Adobe Caslon Pro. I scrounged up a copy from somewhere but I'm not even sure what the legality of the file I have is. It's not readily available, at least not as an Arch Linux Package. We need to either find a legal free source (and teach the build system about it) or replace it with something libre.

### LINKED ISSUE #854: Manual requires fonts not part of build system
Currently cutting releases requires Noto Sans CJK, Gentium Book Basic, and Scheherazade which are all readily available with open licenses (also Adobe Caslon Pro, which is not, see #853); but our build system doesn't actually fetch them so manuals are not consistent unless the builder knows to get these fonts first.

The build system should know that building docs requires these.

### PR DESCRIPTION
**Pull request by simoncozens**
Closes #853
Closes #854

----

This is a WIP which:

- [x] Tidies up the documentation macros.
- [x] Splits the documentation chapters into individual files to make them more manageable.
- [x] Checks and edits all chapters for grammar, style, etc. (Moved to #941)
- [x] Ensures all packages which ship with SILE are fully documented and included in the documentation.
- [x] Ensures that all examples work and that all information is current. (Moved to #943)
- [x] Adds a chapter about SILE's multilingual capabilities.
- [x] Completely reworks the developer-facing documentation. (Moved to #942)

### CODE REVIEW

**Comment by alerque** [comment]
The previous state of this branch had some seriously botched history. Either everything from 2 previous PR's got cherry picked into this branch, master got rebased on top of this branch, or a merge happened and then a rebase flattened it. In any case it had about 60 commits that were not part of the flow of things at all.

I reworked the history, extracted a couple of unrelated commits, dropped the binary blob for Roboto in favor of fetching it as part of the build process, and tweaked a few other font related things.

To get back to a clean state locally (presuming you don't have further local work not pushed as of today) do a `git fetch --all`, switch to this branch `git checkout new-documentation` and hard reset to this repo's branch state `git reset --hard origin/new-documentation`.

Let me know if there is anything else I can do to help this along.

File: documentation/c02-gettingstarted.sil [hunk 1]
```diff
@@ -87,30 +87,31 @@ dependencies as possible. On DEB-based Linux machines such as Debian and
...
 the file that you downloaded from SILE’s home page, change to that directory\footnote{If you downloaded a copy of the SILE source by cloning the git
 repository rather than dowloading one of the release packages you will also
 need to run \code{./bootstrap.sh} to setup the configure script at this point
 before continuing to the next step.}, and run:
 
-\terminal{$ ./configure; make}
+\terminal{$ ./configure && make}
+
+Note that if you prefer and if you system has all the required Lua packages, you
+make opt to skip the download and inclusion of the required Lua rocks and opt to
```
Comment (simoncozens) [inline documentation/c02-gettingstarted.sil]: "may opt"

Comment (alerque) [reply documentation/c02-gettingstarted.sil]: You mean it's not `make opt`? Also "your system" ...

**Comment by simoncozens** [comment]
@alerque Can you help me with the git history here? It's diverged from master and I have no idea how best to get them into sync. Neither rebasing or merging is enjoyable.

**Comment by alerque** [comment]
@simoncozens Sure I have a trick or two that should work for this.

**Comment by alerque** [comment]
Apparently I already started to fix this. I have a branch hanging around called `new-documentation-catastrophe` with some experiments on which tool to merged this with :stuck_out_tongue_winking_eye:

**Comment by alerque** [comment]
I buggered something up with the merge of the Makefiles (probably because it had been merged once already and I duplicated something) so this does not currently build, but the gist of the documentation is correct and I can fix the merge commit later. I need to leave my office for a while so I posted this so I can work on it later from elsewhere.

In the meantime @simoncozens actually feel free to keep working on the documentation files here with new commits on top of this branch. Just don't rebase or merge into here. I'll figure out what the build issue is and fix it separately.

**Comment by alerque** [comment]
Merge issue fixed. The only issue failing the test suite is `tests/package-documentation.sil` — which seems like it might be more related to this PR than my merging mojo.

**Comment by alerque** [comment]
It's not *quite* there, I still have some odds and ends to fix up in the build routines, but when that happens (probably in the next 24 hours) I'm going to want to merge this. As far as I can tell the documentation is in way better shape here than it is in master or any previous release, so I don't see any reason we need to keep mucking around with two sets of files merging edit conflicts between them. I *could* break this out into separate PR's for the copy edit work, the organization changes, the style changes, and the build changes but at this point that's a fair bit of rebasing without much gain.

@simoncozens If there is anything in here you think needs to hold up the show better speak up!

**Pull request merged**